### ISSUE TITLE
Crashes on Bitwig  Linux

### ISSUE THREAD

**Issue by ssabug** [publicsamples/Quetzalcoatl#11]
Hello,

I tried Quetzacoatl vst3 on both Archlinux & Fedora in Bitwig 4.4.

The plugin is successfully scanned by the DAW but on loading the plugin crashes with the error :"plugin host died : Could not read async reply : end of stream"

I had similar errors with the vsts i developped, it was because some files were missing.

I wanted to copy the sample files before running the VST but i didnt found where i am supposed to put them.

**Comment by RickTaylor79** [comment]
Hi, thanks for the report. This is probably related to the other issues this plugin, and [Oi, Grandad](https://github.com/publicsamples/Oi-Grandad/issues/39) have with certain linux distros. It seems okay in Ubuntu/Debian but arch variants have issues. Unfortunately, my linux machine died so it might be some time before a fix is published. 

Unlikely it's down to missing files, as the location is set upon loading plugin for the first time.

**Comment by RickTaylor79** [comment]
Closing due to age.

**Comment by dromer** [comment]
I just tried the `v0.7.1` build and it crashes on Debian stable/12.

In Bitwig it says `Could not read responses: End of stream`

In carla-single we see:

```
X Error of failed request:  BadWindow (invalid Window parameter)
  Major opcode of failed request:  20 (X_GetProperty)
  Resource id in failed request:  0x0
  Serial number of failed request:  54
  Current serial number in output stream:  54
```

**Comment by RickTaylor79** [comment]
Ah, something glitched in the build phase last night..

Just uploaded a fix. Seems to work well in Bitwig demo/Ubuntu 22.

https://github.com/publicsamples/Quetzalcoatl/releases/download/0.7.1/Quetzalcoatl.vst3.linux.tar.xz

Please note that FX are disabled on the linux version for the time being, for some reason the gain levels were crazy with them enabled. They'll be back for the next version..

**Comment by dromer** [comment]
Nope, same failure.

**Comment by RickTaylor79** [comment]
Hmm..  Could you try installing [XSETTINGS](https://packages.debian.org/sid/x11/xsettingsd), please? It should fix this, fingers crossed..

In a terminal, run:

`sudo systemctl enable xsettingsd.service  `

It _should_ report that the service is not found, if this is the case install xsettingsd with:

`sudo apt-get install -y xsettingsd `

Restart & try to load the plugin. It should now work.

**Comment by dromer** [comment]
Hmm, that is a really odd dependency to have. How does your plugin depend on this service to be available?

Installing it from apt does not create any service file, I was able to create an empty configuration in my home directory and start the service by hand and now Quetzalcoatl starts!

I think it would be best if you don't depend on such an odd service for these plugins.
Or is this something that comes from HISE?

**Comment by RickTaylor79** [comment]
Thanks for that. It's not a dependancy as such..

As I understand it, the problem is HISE uses an older, modified version of JUCE that has a bug with desktop environments that aren't compatible with XORG.  XSETTINGS brings X window support to these environments.  See [here](https://github.com/christophhart/HISE/issues/435) for more details. I guess the reason I never ran into this myself is Ubuntu has this baked in. 

Basically, you confirmed what I need to do next. Patch JUCE..

**Comment by dromer** [comment]
Er .. that's weird. I run a classic Xorg window manager (fluxbox) that is fully compliant.

**Comment by RickTaylor79** [comment]
I don't claim to understand what's going here :) Let's see if this patched version does the trick, it's currently baking in the oven.

CMake would open up some interesting possibilities, from what I recall the main dev is pretty opposed to the idea though.  I've toyed with the idea of trying to get HISE running with pamplejuce, but life's too short.

**Comment by RickTaylor79** [comment]
Okay, here a version that should work as the JUCE patch has been applied..

https://github.com/publicsamples/Quetzalcoatl/releases/download/0.7.1/Quetzalcoatl.vst3.linux.Patch.tar.xz

**Comment by dromer** [comment]
Nope, that still gives the same error.

**Comment by RickTaylor79** [comment]
Ugh. Thanks for trying. Maybe I did something wrong? For now, I'm just going to point people to xsettingsd. Install that, use Ubuntu, or build it yourself.  

<insert linux rant here>

**Comment by RickTaylor79** [comment]
Looks like this was [fixed in HISE](https://github.com/christophhart/HISE/commit/669206d8d36d687bd24b369398429401c8d79423) earlier in the year.

v0.8.8 of Quetzalcoatl was built with the latest dev vision, so this should be fixed..

**Comment by dromer** [comment]
Cool, I tried the v0.8.8 build and it loads! (tried with carla-single)

I do see a lot of CPU spikes which occasionally results in massive xruns (dropped frames), but at least we can start playing with it :)

**Comment by dromer** [comment]
Fudge you say? 💩 

I recommend adding Github Actions so builds can happen in CI and others can inspect and chime in ;)

**Comment by RickTaylor79** [comment]
Ya know I'd love to, but I still don't really understand the process. Presumably, Actions doesn't know about JUCE or HISE inherently, so they'd need to be included in the repo I guess? I'll try to get to it this year, but probably with a less complex, more stable project first. 

Next week I'm gonna find a quiet spot and get a grip on this linux thing..
